crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs

```diff
@@ -230,24 +140,41 @@ impl KitsuneP2pActor {
             KitsuneMetrics::count(t, l);
         });
 
-        // wrap in api
-        let f = tx2_api(f, metrics);
+        let mut ep_hnd = None;
+        let mut ep_evt = None;
 
-        // bind local endpoint
-        let ep = f
-            .bind(bind_to, config.tuning_params.implicit_timeout())
-            .await
-            .map_err(KitsuneP2pError::other)?;
+        #[cfg(feature = "tx2")]
+        if ep_hnd.is_none() && config.is_tx2() {
+            tracing::trace!("tx2");
+            let (h, e) = MetaNet::new_tx2(config.clone(), tls_config, metrics).await?;
+            ep_hnd = Some(h);
+            ep_evt = Some(e);
+        }
 
-        // capture endpoint handle
-        let ep_hnd = ep.handle().clone();
+        #[cfg(feature = "tx5")]
+        if ep_hnd.is_none() && config.is_tx5() {
+            tracing::trace!("tx5");
+            let signal_url = match config.transport_pool.get(0).unwrap() {
+                TransportConfig::WebRTC { signal_url } => signal_url.clone(),
+                _ => unreachable!(),
+            };
+            let (h, e) =
+                MetaNet::new_tx5(config.tuning_params.clone(), host.clone(), signal_url).await?;
+            ep_hnd = Some(h);
+            ep_evt = Some(e);
+        }
```

Comment:
Is it desired that the tx2 endpoint is clobbered if both features are enabled?

Reply:
It won't be clobbered because of the `ep_hnd.is_none()` check. So, if tx2 is enabled && the config `is_tx2()` we will get a tx2 hnd.